src/lib/Runner.ts

```diff
@@ -249,21 +252,26 @@ export class Runner {
           await landRequest.update({ dependsOn: null });
           await this.client.stopLandBuild(landRequest.buildId, lockId);
           const user = await this.client.getUser(landRequest.triggererAaid);
-          return landRequest.setStatus('queued', `Queued by ${user.displayName || user.aaid}`);
+          await landRequest.setStatus('queued', `Queued by ${user.displayName || user.aaid}`);
+          return true;
         }
         if (landRequestStatus.state === 'awaiting-merge') {
           const didChangeState = await this.moveFromAwaitingMerge(landRequestStatus, lockId);
           // if we moved, we need to exit early, otherwise, just keep checking the queue
-          if (didChangeState) return this.next();
+          return !!didChangeState;
         } else if (landRequestStatus.state === 'queued') {
-          const movedState = await this.moveFromQueueToRunning(landRequestStatus, lockId);
+          const didChangeState = await this.moveFromQueueToRunning(landRequestStatus, lockId);
           // if the landrequest was able to move from queued to running, exit early, otherwise, keep
           // checking the rest of the queue
-          if (movedState) return this.next();
+          return !!didChangeState;
         }
         // otherwise, we must just be running, nothing to do here
       }
+      return false;
     });
+    if (runNextAgain) {
+      await this.next();
+    }
```

Comment:
Essentially I changed this because I don't think it was causing a lot of the problems to call `next` from inside the locked `next` call. I'm not sure what the specific details are with `awaiting` a recursive call though...